numfun/function.py

```diff
@@ -1447,15 +1452,17 @@ def __init__(self, fun=None, xdata=None, ydata=None, *args, **kwargs):
         if xdata is not None:
             assert ydata is not None, f'xdata is {xdata}, while ydata is {ydata}'
             assert len(xdata) == len(ydata)
-            assert fun is None, f'with xdata and ydata, fun must be None'
+            assert fun is None, 'with xdata and ydata, fun must be None'
```

Comment:
perhaps: 

`assert fun is None, f'type(fun) = {type(fun)} but with xdata and ydata, fun must be None.'`